### ISSUE TITLE
[Crash]: when playing TTO savegame from #5965 due to a regression

### ISSUE THREAD

**Issue by James103** [OpenTTD/OpenTTD#13044]
### Version of OpenTTD

9dae626237569356b1fe25c439edae3c685c49e1 on Windows 10, 64-bit (via Steam)

### Steps to reproduce

Load the attached TTO savegame, then unpause.

`Ship #4` of Draningville Transport near Dintown Docks must be stopped lest it crash the pathfinder and game.
This is because the ship is somehow within the dock's pier tile, not any of the surrounding tiles.

Furthermore, the ship can not be sent to depot because its "unable to find local depot," nor can the dock which the ship is stuck in be cleared because "ship in the way."

This used to work in the release version 14.1, with no crash.

### Upload crash files

[crash20241030074351.zip](https://github.com/user-attachments/files/17568005/crash20241030074351.zip)

**Comment by Kuhnovic** [comment]
Are you sure you uploaded the right savegame? The one provided is not paused and ship 4 seems to move just fine. 

I tested with a local build of master (commit 1b6a6f1cd2f2f5d6a8ff1cd7a526c992b967d3f0 ) and let it run for a while. No crashes, no asserts, everything seemed to run fine.

**Comment by Kuhnovic** [comment]
![image](https://github.com/user-attachments/assets/131d1d77-3165-434d-8f62-f594bf09076e)
Ship 4 indeed seems to be inside the dock in that Trt04.sv1. Ran fine for me, no asserts or errors. (again on commit 1b6a6f1cd2f2f5d6a8ff1cd7a526c992b967d3f0 )

**Comment by Kuhnovic** [comment]
The reason it worked in the official release is because we disable asserts. Nightlies are built with asserts enabled, and so are local builds unless you deliberately turn them off in CMake. Which is what I had done, and then forgot about it...

The triggered assert functions as a sanity check for vehicle movement. It ensures a vehicle can't end up in an invalid location, and is very when working on pathfinders or anything vehicle motion-related. Unfortunately the OTTD saves already contain a vehicle in an invalid state.

**Comment by 2TallTyler** [comment]
Was this ship's position valid in previous versions of OpenTTD/TTD/TTO? If so, we could check vehicle locations when converting old savegames, and teleport any invalid vehicles to the nearest depot, like we did with #9931.

**Comment by James103** [comment]
> Was this ship's position valid in previous versions of OpenTTD/TTD/TTO? If so, we could check vehicle locations when converting old savegames, and teleport any invalid vehicles to the nearest depot, like we did with #9931.

I was able to load the save in OpenTTD version 14.1 with no crash. By waiting a few seconds for Ship 4 to exit the vicinity of the dock and saving the game (still in 14.1), I can load and play the resulting save in the latest nightly (67a0fccfad) with no crash.

As this is the only known savegame I have tested which exhibits this problem, and I have the workaround (described above), I'm closing this issue at least until another similar broken case is found.
